src/state/dimension.js

```diff
@@ -136,8 +143,12 @@ export const getDroppableDimension = ({
 }: GetDroppableArgs): DroppableDimension => {
   const withMargin = getWithSpacing(clientRect, margin);
   const withWindowScroll = getWithPosition(clientRect, windowScroll);
-  // If no containerRect is provided, the clientRect is the container
-  const containerRectWithWindowScroll = getWithPosition(containerRect || clientRect, windowScroll);
+  // If no containerRect is provided, or if the clientRect matches the containerRect, this
+  // droppable is its own container. In that case we include margin in the container dimension.
+  const droppableIsContainer = !containerRect || equal(containerRect, clientRect);
+  const containerRectWithWindowScroll = droppableIsContainer
+    ? getWithPosition(withMargin, windowScroll)
+    : getWithPosition(containerRect || withMargin, windowScroll);
```

Comment:
should we capture the margin of the container, rather than just its clientRect?

Reply:
I think in a case where the container isn't the droppable (i.e. the second or third case in [this diagram](https://user-images.githubusercontent.com/4995433/30788964-4888a78a-a1e5-11e7-8268-14597f2b0f6d.png)) we wouldn't want to know about the container's margin. In the first case we take the droppable's margin into account.